benchmark_runner/common/google_drive/google_drive_operations.py

```diff
@@ -21,33 +22,233 @@ class GoogleDriveOperations:
+
+    def get_folder_id_by_path(self, folder_path, parent_folder_id):
+        """
+        Checks if a folder path exists in Google Drive and returns the ID of the last folder in the path.
+        :param folder_path: The path of folders to check, e.g., '2024/09/15'.
+        :param parent_folder_id: The ID of the parent folder where the search starts.
+        :return: The ID of the last folder in the path if it exists, otherwise None.
+        """
+        folder_names = folder_path.split('/')
+        current_parent_id = parent_folder_id
+
+        for folder_name in folder_names:
+            folder_id = self._find_folder_id(current_parent_id, folder_name)
+            if folder_id:
+                current_parent_id = folder_id
+            else:
+                # Folder in path does not exist
+                return None
+
+        return current_parent_id
+
+    def is_folder_name_exists(self, folder_id, target_folder_name):
+        """
+        Recursively checks if a folder with the specified name exists in a given Google Drive folder.
+        :param folder_id: The ID of the folder to start the search.
+        :param target_folder_name: The name of the folder to find.
+        :return: True if the folder exists, otherwise False.
+        """
+        page_token = None
+        while True:
+            results = self.service.files().list(
+                q=f"'{folder_id}' in parents and trashed = false",
+                includeItemsFromAllDrives=True,
+                supportsAllDrives=True,
+                pageSize=100,
+                fields="nextPageToken, files(id, name, mimeType)",
+                pageToken=page_token
+            ).execute()
+
+            items = results.get('files', [])
+            for item in items:
+                if item['mimeType'] == 'application/vnd.google-apps.folder':
+                    if item['name'] == target_folder_name:
+                        return item['id']
+                    # Recursively check in the subfolders
+                    if self.is_folder_name_exists(item['id'], target_folder_name):
+                        return item['id']
+
+            page_token = results.get('nextPageToken', None)
+            if page_token is None:
+                break
+
+        return False
+
+    def get_drive_folder_url(self, folder_path, parent_folder_id):
+        """
+        Retrieves the Google Drive folder URL based on the folder path without uploading a file.
+        :param folder_path: The folder path to retrieve or create.
+        :param parent_folder_id: The parent folder ID where the folder path begins.
+        :return: The Google Drive URL for the folder.
+        """
+        # Check if the folder exists in the specified path
+        folder_id = self.get_folder_id_by_path(folder_path, parent_folder_id)
+
+        if folder_id:
+            # Folder exists, return the folder URL
+            folder_url = f"https://drive.google.com/drive/folders/{folder_id}"
```

Comment:
You're using this base path in multiple places -- make it a named constant so if it needs to change for some reason you can do it in just one spot.  Maybe

```
def make_google_drive_url(folder_id: str):
    return f"https://drive.google.com/drive/folders/{folder_id}"
```

Reply:
Fixed